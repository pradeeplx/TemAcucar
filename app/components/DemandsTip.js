import React from 'react'
import FacebookTip from "../components/FacebookTip"
import ShareTip from "../components/ShareTip"

export default DemandsTip = ({ currentUser, neighborsCount, onShare, onFacebook, facebookConnecting }) => {
  const title = (neighborsCount < 30 && 'Quanto mais vizinhos, mais trocas! Compartilhe no Facebook, marcando amigos que moram perto de vocÃª.')
  const callToAction = (neighborsCount < 30 && 'Chame seus vizinhos')
  return !currentUser.facebook_uid && neighborsCount >= 30
    ? <FacebookTip onPress={onFacebook} loading={facebookConnecting} title={title} />
    : <ShareTip onPress={onShare} title={title} callToAction={callToAction} />
}
