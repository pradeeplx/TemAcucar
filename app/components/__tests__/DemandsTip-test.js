import { View } from 'react-native'
import React from 'react'
import DemandsTip from '../DemandsTip'
import renderer from 'react-test-renderer'

const props = {
  currentUser: { facebook_uid: 1 },
  neighborsCount: 60,
  facebookConnecting: 'foo',
}

it('renders correctly', () => {
  const tree = renderer.create(
    <DemandsTip {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders different text when neighborsCount < 30', () => {
  const tree = renderer.create(
    <DemandsTip {...props} neighborsCount={29} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders facebook tip when no user uid is set', () => {
  const tree = renderer.create(
    <DemandsTip {...props} currentUser={{}} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});
