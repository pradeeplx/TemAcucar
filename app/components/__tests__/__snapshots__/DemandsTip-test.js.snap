exports[`test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FAF6F1",
        "borderTopWidth": 12,
        "padding": 15,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#000000",
          "fontSize": 15,
          "marginBottom": 10,
          "marginHorizontal": 40,
          "textAlign": "center",
        },
      ]
    }>
    Tá muito feliz e quer espalhar o amor? Conte pra todo mundo que o Tem Açúcar existe!
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#557BD3",
        "borderColor": "#557BD3",
        "borderRadius": 27,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 54,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 0,
        "width": 330,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FF7F85",
              "fontFamily": "Avenir",
              "fontSize": 18,
              "textAlign": "center",
            },
            Array [
              Object {
                "color": "#FFFFFF",
              },
              undefined,
            ],
          ],
        ]
      }>
      Compartilhe no Facebook
    </Text>
  </View>
</View>
`;

exports[`test renders different text when neighborsCount < 30 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FAF6F1",
        "borderTopWidth": 12,
        "padding": 15,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#000000",
          "fontSize": 15,
          "marginBottom": 10,
          "marginHorizontal": 40,
          "textAlign": "center",
        },
      ]
    }>
    Quanto mais vizinhos, mais trocas! Compartilhe no Facebook, marcando amigos que moram perto de você.
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#557BD3",
        "borderColor": "#557BD3",
        "borderRadius": 27,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 54,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 0,
        "width": 330,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FF7F85",
              "fontFamily": "Avenir",
              "fontSize": 18,
              "textAlign": "center",
            },
            Array [
              Object {
                "color": "#FFFFFF",
              },
              undefined,
            ],
          ],
        ]
      }>
      Chame seus vizinhos
    </Text>
  </View>
</View>
`;

exports[`test renders facebook tip when no user uid is set 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FAF6F1",
        "borderTopWidth": 12,
        "padding": 15,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#000000",
          "fontSize": 15,
          "marginBottom": 10,
          "marginHorizontal": 40,
          "textAlign": "center",
        },
      ]
    }>
    Quer ter acesso a funcionalidades incríveis?
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#557BD3",
        "borderColor": "#557BD3",
        "borderRadius": 27,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 54,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 0,
        "width": 330,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FF7F85",
              "fontFamily": "Avenir",
              "fontSize": 18,
              "textAlign": "center",
            },
            Array [
              Object {
                "color": "#FFFFFF",
              },
              undefined,
            ],
          ],
        ]
      }>
      Conecte sua conta do Facebook
    </Text>
  </View>
</View>
`;
